packages/xo-server/src/xo-mixins/store.js

```diff
@@ -1,29 +1,84 @@
+import assert from 'assert'
 import levelup from 'level-party'
 import sublevel from 'level-sublevel'
+import { createPredicate } from 'value-matcher'
+import { defer, fromEvent } from 'promise-toolbox'
 import { ensureDir } from 'fs-extra'
 
 import { forEach, promisify } from '../utils'
 
 // ===================================================================
 
-const _levelHas = function has(key, cb) {
-  if (cb) {
-    return this.get(key, (error, value) =>
-      error ? (error.notFound ? cb(null, false) : cb(error)) : cb(null, true)
-    )
-  }
+async function _levelHas(key) {
+  return this.get(key).then(
+    () => true,
+    error => {
+      if (!error.notFound) {
+        throw error
+      }
+      return false
+    }
+  )
+}
+
+// keep n element in the DB
+async function _levelGc(keep) {
+  assert(typeof keep === 'number')
 
-  try {
-    this.get(key)
-    return true
-  } catch (error) {
-    if (!error.notFound) {
-      throw error
+  let count = 1
+  const { promise, resolve } = defer()
+
+  const cb = () => {
+    if (--count === 0) {
+      resolve()
     }
   }
-  return false
+  const stream = this.createKeyStream({
+    reverse: true,
+  })
+
+  const deleteEntry = key => {
+    ++count
+    this.del(key, cb)
+  }
+
+  const onData =
+    keep !== 0
+      ? () => {
+          if (--keep === 0) {
+            stream.on('data', deleteEntry)
+            stream.removeListener('data', onData)
+          }
+        }
+      : deleteEntry
+  stream.on('data', onData)
+
+  await fromEvent(stream, 'end')
+  cb()
+
+  return promise
+}
+
+function _levelGetAll(filter = {}) {
+  const isMatch = createPredicate(filter)
```

Comment:
Use `lodash/iteratee` instead, we don't need the features of `value-matcher` and we might want to pass a function directly for performance.